### DISCUSSION TITLE
file is not a module for supabase generated types file

### DISCUSSION THREAD

**Discussion by esskayesss** [nuxt/nuxt#21856] [Questions]
i am working on a nuxtjs3 project which also happens to use supabase. i generated the types for the database tables using the following command, as is talked about in the docs:
```
npx supabase gen types typescript --project-id "<id>" --schema public > types/supabase.ts
```

when i try to import `Database` from the file like:
```js
import { Database } from '@/types/supabase';
```
it throws an error:
```
File 'C:/Users/.../<project>/types/supabase.ts' is not a module.ts(2306)
```

clearly this isn't a file resolution error as the correct file is being complained about. 
the `types/supabase.ts` file does correctly export everything:
```js
export type Json =
  | string
  | number
  | boolean
  | null
  | { [key: string]: Json | undefined }
  | Json[]

export interface Database {
  public: {
    Tables: {
      //...
    }
  }
}
```

i want the types to be imported. i've tried getting help from GPT, and i've also tried changing the file aliases even though it clearly isn't an issue as i've talked about in the post.

**Comment by 0xBoom** [comment]
I found out that the generated file was encoded in UTF-16, changing to UTF-8 solved the issue for me.

in Powershell I did:
```ps
Get-Content src/supabase/database.types.ts | 
Set-Content -Encoding UTF8 src/supabase/database.types.ts.new | 
Move-Item -Force src/supabase/database.types.ts.new src/supabase/database.types.ts;
```
